Automattic/wp-calypso
# Stats: unify purchases loading for Odyssey and Calypso

## Proposed Changes

* Created a Odyssey implementation `OdysseyQuerySitePurchases` of 'calypso/components/data/query-site-purchases' and replace at build time
* <strike>Extracted product tests to a hook `useStatsPurchases`, which could be used in Traffic page banner visibility ( cc @annacmc ) and the enabling of the purchase page to Odyssey</strike>

## Testing Instructions

Calypso Stats
* Inspect network requests
* Ensure there's a request sent to `/rest/v1.1/sites/:siteId/purchases`
* Ensure redux state `purchases` is hydrated

Odyssey Stats
* Inspect network requests
* Ensure there's a request sent to `/wp-json/jetpack/v4/site/purchases`
* Ensure redux state `purchases` is hydrated

Both
* Ensure purchases still works for Odyssey and Calypso Stats
  * Notices on Traffic page still work  as expected
  * Purchase still work as expected


## Pre-merge Checklist
